### PR TITLE
ApiConventionAnalyzer: support switch expressions

### LINKED ISSUE #42987: `ApiConventionAnalyzer` does not work for `switch` expressions
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using `switch` expressions to return an `IActionResult`, `ApiConventionAnalyzer` is unable to detect undeclared status codes.

### Expected Behavior

_No response_

### Steps To Reproduce

```cs
[ApiController]
public class Foo : ControllerBase
{
    public IActionResult Get(bool b)
    {
        if (b)
        {
            return Ok();
        }
        else
        {
            return BadRequest(); // Produces warning API1000 as expected.
        }
    }
}
```

```cs
[ApiController]
public class Foo : ControllerBase
{
    public IActionResult Get(bool b)
    {
        return b switch
        {
            true => Ok(),
            false => BadRequest() // Does not produce warning API1000.
        };
    }
}
```

### Exceptions (if any)

_No response_

### .NET Version

7.0.100-preview.6.22352.1

### Anything else?

AspNetCore.OpenApi version: `7.0.0-preview.6.22330.3`  
AspNetCore.Mvc.Api.Analyzers version: `2.2.6`  
Visual Studio version: `17.3.0 Preview 5.0`  

`dotnet --info` output:  
```
.NET SDK:
 Version:   7.0.100-preview.6.22352.1
 Commit:    492644e08e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-preview.6.22352.1\

Host:
  Version:      7.0.0-preview.6.22324.4
  Architecture: x64
  Commit:       d3fa592f6d

.NET SDKs installed:
  6.0.400-preview.22330.6 [C:\Program Files\dotnet\sdk]
  7.0.100-preview.6.22352.1 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.6.22330.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.6.22324.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.6.22351.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
```

### PR DESCRIPTION
**Pull request by david-acker**
# ApiConventionAnalyzer: support switch expressions

## Description

Updates `ApiConventionAnalyzer` to correctly handle return operations using switch expressions. 

Fixes #42987

### CODE REVIEW

File: src/Mvc/Mvc.Api.Analyzers/test/ActualApiResponseMetadataFactoryTest.cs [hunk 1]
```diff
@@ -14,6 +14,12 @@ public class ActualApiResponseMetadataFactoryTest
 {
     private static readonly string Namespace = typeof(ActualApiResponseMetadataFactoryTest).Namespace;
 
+    public enum ReturnOperationTestVariant
```
Comment (david-acker) [inline src/Mvc/Mvc.Api.Analyzers/test/ActualApiResponseMetadataFactoryTest.cs:17]: I opted to use an enum to differentiate between the types of test cases, in order to make things a bit clearer, but if this seems overkill I can change this to a boolean instead.

File: src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs [hunk 2]
```diff
@@ -53,16 +56,17 @@ void AnalyzeResponseExpression(IReturnOperation returnOperation)
...
+        ISwitchExpressionArmOperation? armOperation = null)
     {
-        var returnedValue = returnOperation.ReturnedValue;
+        var returnedValue = armOperation?.Value ?? returnOperation.ReturnedValue;
         var defaultStatusCodeAttributeSymbol = symbolCache.DefaultStatusCodeAttribute;
 
         if (returnedValue is null || returnedValue is IInvalidOperation)
         {
-            return null;
+            return new ActualApiResponseMetadata?[] { null };
```
[the remark this answers is not in the document]
Comment (david-acker) [reply src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs]: Ah nice! I've updated these to use collection expressions.

File: src/Mvc/Mvc.Api.Analyzers/test/TestFiles/ActualApiResponseMetadataFactoryTest/InspectReturnExpressionTests.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@ namespace Microsoft.AspNetCore.Mvc.Api.Analyzers.TestFiles.InspectReturnExpressi
 {
     public class TestController : ControllerBase
     {
+        #region Default
```
[the remark this answers is not in the document]
Comment (david-acker) [reply src/Mvc/Mvc.Api.Analyzers/test/TestFiles/ActualApiResponseMetadataFactoryTest/InspectReturnExpressionTests.cs]: Ah good to know. I've moved the switch expression test cases into a separate test file. Makes the method names a bit shorter too.

**Review by captainsafia — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
Thanks @david-acker!

**Pull request merged**